### DISCUSSION TITLE
Show car's battery state of health (SOH)

### DISCUSSION THREAD

**Discussion by eerwinn** [flobz/psa_car_controller#839] [Ideas]
I'm not sure if this possible, or already in place, but a would be great to keep track on battery degradation. (SOH)
Extraction below is coming from a 508 HYbrid. If the health values  **(in bold)**  would represent  the SOH of the battery then maybe then the data should be there.

  "energy": [
    {
      "updated_at": "2024-03-20 17:12:21+00:00",
      "autonomy": 28,
      "battery": {
        "capacity": null,
        "health": {
          **"capacity": ??,**
          "resistance": 0
        }
      },
      "charging": {
        "charging_mode": "No",
        "charging_rate": 0,
        "next_delayed_time": "PT0S",
        "plugged": false,
        "remaining_time": null,
        "status": "Disconnected"
      },
      "consumption": null,
      "level": 90,
      "residual": null,
      "type": "Electric"
    },
    {
      "updated_at": "2024-03-20 17:12:21+00:00",
      "autonomy": 125,
      "battery": null,
      "charging": null,
      "consumption": null,
      "level": 21,
      "residual": null,
      "type": "Fuel"
    }
  ]

**Comment by malbertone** [comment]
Hi,
think you can use the specific API for that; based on:
https://github.com/flobz/psa_car_controller/blob/master/docs/psacc_api.md
Get battery SOH
http://localhost:5000/battery/soh/YOURVIN
the output will be something like:
![image](https://github.com/flobz/psa_car_controller/assets/36228023/349e20b3-8fdf-4782-9aa7-b1bd08885918)
And you can define your sensor like:
    - resource: "http://127.0.0.1:5000/battery/soh/YOURVIN?from_cache=1" 
    scan_interval: 5
    sensor:
      - name: PSA_SOH
        unique_id: psa_soh
        value_template: 'OK'
        json_attributes:
          - soh

**Reply by eerwinn** [reply]
Hi @malbertone,

If I use the http://localhost:5000/battery/soh/YOURVIN directly the output is always:
{"soh":0.0}

If I check the output from: http://localhost:5000/get_vehicleinfo/YOURVIN?from_cache=1  or without cache..
Then I do get information but it seems that the information is not always there.

When Charging: value is present.
When Finished Charging: value is not present.
When Charger is disconnected: value is not present.

this template is reflecting the value in **bold**. Although I am not sure if this is the same as the one your describe above. That one is always on **{"soh":0.0}**.
'{{ states.sensor.peugeot_508.attributes["energy"][0]["battery"]["health"]["capacity"] }}'

  },
  "battery": {
    "current": null,
    "voltage": 100
  },
  "doors_state": null,
  "energy": [
    {
      "updated_at": "2024-04-11 08:59:59+00:00",
      "autonomy": 30,
      "battery": {
        "capacity": null,
        "health": {
          **"capacity": 100,**
          "resistance": 0
        }
      },
      "charging": {
        "charging_mode": "No",
        "charging_rate": 0,
        "next_delayed_time": "PT8H15M",
        "plugged": false,
        "remaining_time": null,
        "status": "Disconnected"
      },
      "consumption": null,
      "level": 100,
      "residual": null,
      "type": "Electric"
    },
    {
      "updated_at": "2024-04-11 09:00:51+00:00",
      "autonomy": 560,
      "battery": null,
      "charging": null,
      "consumption": 0,
      "level": 74,
      "residual": null,
      "type": "Fuel"
    }
  ]

**Reply by malbertone** [reply]
Hi, think it highly depends on the car's model and the APIs implemented by PSA. I have a 508 SW Hybrid made on 2023, and in my case i don't see for instance the capacity item, while i see the soh being 100% in the specific internal API.
Cannot comment on the charging data since have the car in a garage without connectivity, but not that confident about PSA APIs sending most of the information they could, since tried to monitor the kinetic items and never saw it being updated for instance even when the car is in motion (I was not the driver :))
So long story short, it seems to be a trial and error, so if that value works for you it's great. Have to say that initially in the PSA Controller page on HA did not see the SOH in place, but now it's there:
![image](https://github.com/flobz/psa_car_controller/assets/36228023/8e5c98c9-69ac-49f5-8d35-6fd0e4b1facd)
so think maybe PSA started sending the data and PSA Controller properly reported it internally.
Cheers
Mike

**Reply by eerwinn** [reply]
I have the 2020 model 508 HYbrid. I presume you have the new 508 (facelift) and that one has a different firmware I assume.

**Comment by zadow78** [comment]
> I'm not sure if this possible, or already in place, but a would be great to keep track on battery degradation. (SOH) Extraction below is coming from a 508 HYbrid. If the health values **(in bold)** would represent the SOH of the battery then maybe then the data should be there.
> 
> "energy": [ { "updated_at": "2024-03-20 17:12:21+00:00", "autonomy": 28, "battery": { "capacity": null, "health": { **"capacity": ??,** "resistance": 0 } }, "charging": { "charging_mode": "No", "charging_rate": 0, "next_delayed_time": "PT0S", "plugged": false, "remaining_time": null, "status": "Disconnected" }, "consumption": null, "level": 90, "residual": null, "type": "Electric" }, { "updated_at": "2024-03-20 17:12:21+00:00", "autonomy": 125, "battery": null, "charging": null, "consumption": null, "level": 21, "residual": null, "type": "Fuel" } ]

not sure weather you have solved in the meantime, but my C5 PHEV gives exactly the same output for SOH. This is the sensor what I used to read it and display in HomeAssistant:

ec4_soh_status:
      friendly_name: "Battery SOH Capacity"
      unit_of_measurement: "%"
      value_template: "{{ state_attr('sensor.citroen_ec4', 'energy')[0]['battery']['health']['capacity'] }}"
      icon_template: mdi:battery-heart
